src/lightning/pytorch/strategies/deepspeed.py

```diff
@@ -782,6 +782,19 @@ def load_checkpoint(self, checkpoint_path: _PATH) -> Dict[str, Any]:
         from lightning.pytorch.trainer.states import TrainerFn
 
         is_fitting = self.lightning_module.trainer.state.fn == TrainerFn.FITTING
+
+        # Determine the parent folder of the checkpoint_path
+        parent_folder = os.path.dirname(checkpoint_path)
+
+        # Check if the parent folder is a valid deepspeed checkpoint directory
+        is_deepspeed_checkpoint_dir = os.path.isdir(parent_folder) and "checkpoint" in os.listdir(parent_folder)
```

Comment:
Looks great! Let's put these two lines in a helper function. We would also need the same check in fabric (src/fabric/strategies/deepspeed.py), so we could move the function there and import it here. Sounds good?

Reply:
Cool, I am on it!

Reply:
I have updated the code, however I see that the name of the checkpoint directory within the parent directory is not just `checkpoint` but it is varying. 

Is it supposed to be like that? If yes then instead of ensuring that the `checkpoint` folder exists in the parent directory I will check that the any folder having `checkpoint` in its name exists in the parent directory.